Rename message_command variables in src/net* and src/rpc/net.cpp

shaavan opened the pull request:
This PR is a follow-up to #24078.

> a message is not a command, but simply a message of some type

The first commit covers the message_command variable name and comments not addressed in the original PR in `src/net*` files.
The second commit goes beyond the original `src/net*` limit of #24078 and does similar changes in the `src/rpc/net.cpp` file.

hebasto commented:
Concept ACK. Making commits a "scripted-diff" will simplify reviewing a lot.

prusnak commented:
> Concept ACK. Making rename commits a "scripted-diff" will simplify reviewing a lot.

Link to documentation how to use `scripted-diff` https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#scripted-diffs

hebasto commented:
> ... and random commit showing how the commit message should look like: [eb04bad](https://github.com/bitcoin/bitcoin/commit/eb04badcd6108e17c8ec78e9c316775c334030cb)

Another example -- c4a31ca267f74bff76a43878177d05d22825a203 -- with `git grep --files-with-matches` usage.

RandyMcMillan commented:
I ran this script against your PR - https://github.com/shaavan/bitcoin/blob/4b119e3b95a42f0b6eb3411f961bac50ea225d58/24141-scripted-diff.sh

```
#!/bin/bash -e

function s1() { git grep -l "$1" src/net.h | xargs sed -i .old "s/$1/$2/g"; }

 s1 'mapSendBytesPerMsgCmd' 'mapSendBytesPerMsg'
 s1 'NET_MESSAGE_COMMAND_OTHER' 'NET_MESSAGE_OTHER'
 s1 'mapRecvBytesPerMsgCmd' 'mapRecvBytesPerMsg'
 s1 'mapMsgCmdSize' 'mapMsgSize'
 s1 'sendPerMsgCmd' 'sendPerMsg'
 s1 'mapSendBytesPerMsgCmd' 'mapSendBytesPerMsg'
 s1 'recvPerMsgCmd' 'recvPerMsg'

function s2() { git grep -l "$1" src/net.cpp | xargs sed -i .old "s/$1/$2/g"; }

 s2 'mapSendBytesPerMsgCmd' 'mapSendBytesPerMsg'
 s2 'NET_MESSAGE_COMMAND_OTHER' 'NET_MESSAGE_OTHER'
 s2 'mapRecvBytesPerMsgCmd' 'mapRecvBytesPerMsg'
 s2 'mapMsgCmdSize' 'mapMsgSize'
 s2 'sendPerMsgCmd' 'sendPerMsg'
 s2 'mapSendBytesPerMsgCmd' 'mapSendBytesPerMsg'
 s2 'recvPerMsgCmd' 'recvPerMsg'

function s3() { git grep -l "$1" src/rpc/net.cpp | xargs sed -i .old "s/$1/$2/g"; }

 s3 'mapSendBytesPerMsgCmd' 'mapSendBytesPerMsg'
 s3 'NET_MESSAGE_COMMAND_OTHER' 'NET_MESSAGE_OTHER'
 s3 'mapRecvBytesPerMsgCmd' 'mapRecvBytesPerMsg'
 s3 'mapMsgCmdSize' 'mapMsgSize'
 s3 'sendPerMsgCmd' 'sendPerMsg'
 s3 'mapSendBytesPerMsgCmd' 'mapSendBytesPerMsg'
 s3 'recvPerMsgCmd' 'recvPerMsg'
```

RandyMcMillan commented:
Concept ACK

w0xlt commented on src/rpc/net.cpp:
```diff
@@ -156,7 +156,7 @@ static RPCHelpMan getpeerinfo()
                         {RPCResult::Type::NUM, "msg", "The total bytes received aggregated by message type\n"
                                                       "When a message type is not listed in this json object, the bytes received are 0.\n"
                                                       "Only known message types can appear as keys in the object and all bytes received\n"
-                                                      "of unknown message types are listed under '"+NET_MESSAGE_COMMAND_OTHER+"'."}
```
Shouldn't this be in the previous commit?

shaavan replied:
This change is done in src/rpc/net.cpp file. The first commit makes changes to src/net.cpp and src/net.h files.

w0xlt commented on src/net.cpp:
```diff
@@ -605,12 +605,12 @@ void CNode::CopyStats(CNodeStats& stats)
     {
         LOCK(cs_vSend);
-        X(mapSendBytesPerMsgCmd);
+        X(mapSendBytesPerMsg);
```
Shouldn't `mapSendBytesPerMsgCmd` be renamed to `mapSendBytesPerMsgType`?

shaavan replied:
I think this would be a better naming convention for the variables.

Addressed this and the following comments in 8f31aadd25ad434536ba0d166b81324a18180acb

w0xlt commented on src/net.cpp:
```diff
@@ -102,7 +102,7 @@ enum BindFlags {
 static const uint64_t SELECT_TIMEOUT_MILLISECONDS = 50;
 
-const std::string NET_MESSAGE_COMMAND_OTHER = "*other*";
+const std::string NET_MESSAGE_OTHER = "*other*";
```
Shouldn't `NET_MESSAGE_COMMAND_OTHER` be renamed to `NET_MESSAGE_TYPE_OTHER`?

w0xlt commented on src/net.cpp:
```diff
@@ -605,12 +605,12 @@ void CNode::CopyStats(CNodeStats& stats)
     {
         LOCK(cs_vRecv);
-        X(mapRecvBytesPerMsgCmd);
+        X(mapRecvBytesPerMsg);
```
Shouldn't `mapRecvBytesPerMsgCmd` be renamed to `mapRecvBytesPerMsgType`?

w0xlt commented on src/net.h:
```diff
@@ -233,8 +233,8 @@ struct LocalServiceInfo {
-extern const std::string NET_MESSAGE_COMMAND_OTHER;
-typedef std::map<std::string, uint64_t> mapMsgCmdSize; //command, total bytes
+extern const std::string NET_MESSAGE_OTHER;
+typedef std::map<std::string, uint64_t> mapMsgSize; //command, total bytes
```
Shouldn't `mapMsgCmdSize` be renamed to `mapMsgTypeSize` ?

RandyMcMillan replied:
I am going to disagree - mapMsgSize seems correct (IMO)

w0xlt commented on src/rpc/net.cpp:
```diff
@@ -243,19 +243,19 @@ static RPCHelpMan getpeerinfo()
 
-        UniValue recvPerMsgCmd(UniValue::VOBJ);
-        for (const auto& i : stats.mapRecvBytesPerMsgCmd) {
+        UniValue recvPerMsg(UniValue::VOBJ);
```
Shouldn't `recvPerMsgCmd` be renamed to `recvPerMsgType` ?

w0xlt commented on src/rpc/net.cpp:
```diff
@@ -243,19 +243,19 @@ static RPCHelpMan getpeerinfo()
 
-        UniValue sendPerMsgCmd(UniValue::VOBJ);
-        for (const auto& i : stats.mapSendBytesPerMsgCmd) {
+        UniValue sendPerMsg(UniValue::VOBJ);
```
Shouldn't `sendPerMsgCmd` be renamed to `sendPerMsgType` ?

w0xlt reviewed (commented):
Approach ACK

RandyMcMillan commented:
After some consideration - I generally agree with @w0xlt - the values held in the string are defined in 

https://github.com/bitcoin/bitcoin/blob/9ec3991ad3b2ae91997c696a4c2f187fe538eff0/src/protocol.h#L64

&

https://github.com/bitcoin/bitcoin/blob/9ec3991ad3b2ae91997c696a4c2f187fe538eff0/src/protocol.cpp#L12

The `Cmd` part of the naming convention seems to be taken from COMMAND_SIZE originally defined here -> https://github.com/bitcoin/bitcoin/blame/c729dbb6d2b63d9c0b32e3937be0ebf528d4c753/src/protocol.h#L52

shaavan commented:
Updated from e80330b to 8f31aad ([pr24141.01](https://github.com/shaavan/bitcoin/tree/pr24141.01) -> [pr24141.02](https://github.com/shaavan/bitcoin/tree/pr24141.02))
Addressed @hebasto @w0xlt suggestions.

Changes:
- Renamed commit message to a scripted-diff message.
- Renamed variables from `message_command` -> `message_type` instead of `message_command` -> `message`.

Thanks, @prusnak and @hebasto, for the scripted-diff example and documentation. And thanks, @RandyMacmillian, for the script message, which I have used as an inspiration for the commit message.

w0xlt reviewed (approved):
ACK 8f31aad

Lint is failing because the changed comments do not match the pattern of the sed command.

Change the comments in another commit and this CI error will be fixed.

RandyMcMillan commented:
@w0xlt - Interesting nuance to scripted diffs - thanks for the insight. 😄 - basically each change needs its own commit in incremental steps.

shaavan commented:
> Lint is failing because the changed comments do not match the pattern of the sed command.

Nice catch, @w0xlt! I reckoned the comment change is relatively small and wouldn't require a separate commit. But let's keep the lint intact. Let me update the PR with separating comment changes in a new commit.

shaavan committed:
refactor: rename command -> message type in comments in the src/net* files

Co-authored-by: Sebastian Falbesoner <sebastian.falbesoner@gmail.com>

shaavan commented:
Updated from 8f31aad to d077901 ([pr24141.02](https://github.com/shaavan/bitcoin/tree/pr24141.02) -> [pr24141.03](https://github.com/shaavan/bitcoin/tree/pr24141.03))
Addressed @w0xlt suggestion.

Changes:
- Separated comment changes in a separate commit, allowing lint test to pass successfully on scripted-diff commits.

w0xlt reviewed (approved):
reACK d077901

MarcoFalke commented:
Does each commit compile on its own?

shaavan commented:
Updated from d077901 to ec50abf ([pr24141.03](https://github.com/shaavan/bitcoin/tree/pr24141.03) -> [pr24141.04](https://github.com/shaavan/bitcoin/tree/pr24141.04))
Addressed @MarcoFalke comment.

Changes:
- Merged scripted diff commits (commit f85a0fca0e41260ac43f6e093b63f49084ba335a and d077901f4d4b4b7e6477ef4326c76aebdeba4a46) together. This was done to make the individual commits compilable.

Thanks, @MarcoFalke, for catching it.

theStack reviewed (commented):
Concept ACK

Some more candidates for the second commit:
```diff
diff --git a/src/net.cpp b/src/net.cpp
index f3ae11ce1..04d9b318d 100644
--- a/src/net.cpp
+++ b/src/net.cpp
@@ -748,7 +748,7 @@ CNetMessage V1TransportDeserializer::GetMessage(const std::chrono::microseconds
     // decompose a single CNetMessage from the TransportDeserializer
     CNetMessage msg(std::move(vRecv));

-    // store command string, time, and sizes
+    // store message type string, time, and sizes
     msg.m_type = hdr.GetCommand();
     msg.m_time = time;
     msg.m_message_size = hdr.nMessageSize;
@@ -759,7 +759,7 @@ CNetMessage V1TransportDeserializer::GetMessage(const std::chrono::microseconds
     // We just received a message off the wire, harvest entropy from the time (and the message checksum)
     RandAddEvent(ReadLE32(hash.begin()));

-    // Check checksum and header command string
+    // Check checksum and header message type string
     if (memcmp(hash.begin(), hdr.pchChecksum, CMessageHeader::CHECKSUM_SIZE) != 0) {
         LogPrint(BCLog::NET, "Header error: Wrong checksum (%s, %u bytes), expected %s was %s, peer=%d\n",
                  SanitizeString(msg.m_type), msg.m_message_size,
```

```diff
diff --git a/src/net.h b/src/net.h
index b1a19bcc4..62d23f6c4 100644
--- a/src/net.h
+++ b/src/net.h
@@ -298,7 +298,7 @@ public:

 /** The TransportDeserializer takes care of holding and deserializing the
  * network receive buffer. It can deserialize the network buffer into a
- * transport protocol agnostic CNetMessage (command & payload)
+ * transport protocol agnostic CNetMessage (message type & payload)
  */
 class TransportDeserializer {
 public:
```

```diff
diff --git a/src/net_processing.cpp b/src/net_processing.cpp
index cc23876af..3bcf2d1b1 100644
--- a/src/net_processing.cpp
+++ b/src/net_processing.cpp
@@ -4078,8 +4078,8 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         return;
     }

-    // Ignore unknown commands for extensibility
-    LogPrint(BCLog::NET, "Unknown command \"%s\" from peer=%d\n", SanitizeString(msg_type), pfrom.GetId());
+    // Ignore unknown messages types for extensibility
+    LogPrint(BCLog::NET, "Unknown message type \"%s\" from peer=%d\n", SanitizeString(msg_type), pfrom.GetId());
     return;
 }
```

(Though changing a log message is not a refactor strictly speaking, so not sure if it's appropriate in this PR.)

shaavan commented:
Updated from https://github.com/bitcoin/bitcoin/commit/ec50abf1d0666706093a004773cbfa5229c6f618 to 15eb7b94965827bb4e1ac11af9fc61fcbf044cd5 ([pr24141.04](https://github.com/shaavan/bitcoin/commits/pr24141.04) → [pr24141.05](https://github.com/shaavan/bitcoin/commits/pr24141.05), [diff](https://github.com/shaavan/bitcoin/compare/pr24141.04..pr24141.05))

Addressed @theStack comments.

Included the suggestions suggested by @theStack [here](https://github.com/bitcoin/bitcoin/pull/24141#pullrequestreview-870469607).

**About the Log message:**
I thought about adding it in the 2nd commit, but then I inclined against it for the following reason:

1. This will not strictly be a refactoring change
2. This changes a file (net_processing.cpp) untouched in this PR.

But I think changing the log message would be essential to make it consistent with the other changes done in this PR. So I would require other reviewers’ suggestions on this matter, on whether it is better to edit the log message or not.

theStack reviewed (approved):
Code-review ACK 15eb7b94965827bb4e1ac11af9fc61fcbf044cd5

> **About the Log message:** I thought about adding it in the 2nd commit, but then I inclined against it for the following reason:
> 
> 1. This will not strictly be a refactoring change
> 2. This changes a file (net_processing.cpp) untouched in this PR.
> 
> But I think changing the log message would be essential to make it consistent with the other changes done in this PR. So I would require other reviewers’ suggestions on this matter, on whether it is better to edit the log message or not.

Probably it could be done with another follow-up (final) PR which replaces the remaining occurences of the "message command" terminology in src/protocol.{h,cpp} and src/net_processing.cpp?

MarcoFalke commented:
Not sure about the scripted diff. It seems a bit bloated and throws errors (https://cirrus-ci.com/task/6681881558122496?logs=lint#L842). Try something like:

```
    scripted-diff: Rename message command to message type
    
    -BEGIN VERIFY SCRIPT-
    
     s1() { sed -i "s/$1/$2/g" $(git grep -l "$1" ./); }
    
     s1 'NET_MESSAGE_COMMAND_OTHER' 'NET_MESSAGE_TYPE_OTHER'
     s1 'mapMsgCmdSize' 'mapMsgTypeSize'
     s1 'mapRecvBytesPerMsgCmd' 'mapRecvBytesPerMsgType'
     s1 'mapSendBytesPerMsgCmd' 'mapSendBytesPerMsgType'
     s1 'recvPerMsgCmd' 'recvPerMsgType'
     s1 'sendPerMsgCmd' 'sendPerMsgType'
    
    -END VERIFY SCRIPT-

laanwj commented:
Concept ACK

MarcoFalke commented:
Are you still working on this? https://github.com/bitcoin/bitcoin/pull/24141#issuecomment-1061834310

shaavan committed:
scripted-diff: Rename message command to message type

-BEGIN VERIFY SCRIPT-

 s1() { sed -i "s/$1/$2/g" $(git grep -l "$1" ./); }

 s1 'NET_MESSAGE_COMMAND_OTHER' 'NET_MESSAGE_TYPE_OTHER'
 s1 'mapMsgCmdSize' 'mapMsgTypeSize'
 s1 'mapRecvBytesPerMsgCmd' 'mapRecvBytesPerMsgType'
 s1 'mapSendBytesPerMsgCmd' 'mapSendBytesPerMsgType'
 s1 'recvPerMsgCmd' 'recvPerMsgType'
 s1 'sendPerMsgCmd' 'sendPerMsgType'

-END VERIFY SCRIPT-

MarcoFalke commented on src/net.h:
```diff
@@ -233,8 +233,8 @@ struct LocalServiceInfo {
-extern const std::string NET_MESSAGE_COMMAND_OTHER;
-typedef std::map<std::string, uint64_t> mapMsgCmdSize; //command, total bytes
+extern const std::string NET_MESSAGE_TYPE_OTHER;
+typedef std::map<std::string, uint64_t> mapMsgTypeSize; //message, total bytes
```
nit: Second commit `s/message/message type/`. Or:

```suggestion
using mapMsgTypeSize = std::map</* msg type */ std::string, /* total bytes */ uint64_t>;
```

shaavan replied:
Thanks, @MarcoFalke, for taking another look. As per your suggestion, I was able to identify one more place in the second commit where your suggestion could be implemented.

```diff
-            // to prevent a memory DOS, only allow valid commands
+            // to prevent a memory DOS, only allow valid messages
```
But I opted against changing the above from `messages` to `message types`, as that would have decreased the sentence's logical soundness IMO.

maflcko reviewed (approved):
ACK

shaavan commented:
Updated from [pr24141.06](https://github.com/shaavan/bitcoin/commits/pr24141.06) → [pr24141.07](https://github.com/shaavan/bitcoin/commits/pr24141.07)

Addressed @MarcoFalke suggestion.

Changes:

- Replaced usage of `typedef` with `using` in defining mapMsgTypeSize.
- Replaced `message` with `message type` in the comment defining the above.

MarcoFalke commented on src/net.cpp:
```diff
@@ -651,14 +651,14 @@ bool CNode::ReceiveMsgBytes(Span<const uint8_t> msg_bytes, bool& complete)
-            // Store received bytes per message command
-            // to prevent a memory DOS, only allow valid commands
+            // Store received bytes per message type
+            // to prevent a memory DOS, only allow valid messages
```
I think this should say "known message types"?

For example, a `reject` message might be valid according to BIP 61, but unknown to this version of Bitcoin Core.

shaavan replied:
Thanks for seeing it. I think you are right. Using "known message types" would be a better wording for this sentence.
Let me address it by updating the second commit.

jonatack commented:
Approach ACK

shaavan commented:
Update from [pr24141.07](https://github.com/shaavan/bitcoin/commits/pr24141.07) → [pr24141.08](https://github.com/shaavan/bitcoin/commits/pr24141.08), [diff](https://github.com/shaavan/bitcoin/compare/pr24141.07..pr24141.08)

Addressed suggestion by @MarcoFalke

Changes:

- Change the wording of the comment from “_allow valid messages_” to “_allow known message types_” as some of the messages (like `reject`, as pointed out by @MarcoFalke, may be valid but yet unknown to the current version of bitcoin core)
- Also did some punctuation corrections to make the comment grammatically sound.

maflcko reviewed (approved):
review ACK e71c51b27d420fbd6cc0a36f62e63e190e13473a 💥

<details><summary>Show signature</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

review ACK e71c51b27d420fbd6cc0a36f62e63e190e13473a 💥
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhJ/Qv+KgTz8htAqBuJ6FnTHQdjyLswJvYHXukDceneF5E72Gifx7MaJw439PKt
MjVpE6s1VCZIliOhSojkNrINaL07pLQu+MoDNSkdOHexfm6xlsDg29nBdba0ocLq
meTNnDW+f+uUibpz+6NxY7Fur8YIpyz+Ney8X0EkOa73F01Dcy9rjAzRZRjCI/gk
X9i7kWzByG7uTzAAvuNxV+F1Kg1qVX2GgMOtxmYj+qpbx2kVCd7YA2pUjYbLeuPL
9/SWmNdYnjFPznUUhby0HcSY0wwZ5urKIhWPgVeXpQgnQJu4WCNYum+ckBIThhor
uS1IkAtylgZfwcEU4m8unW4dUoQkeCFyN8xVSsQztL1TinM3CmzKnsgZq1LkU2El
i1HTy5qxZCybGh2y3R/bUuygFelDw9PZY2Qm/AtvCaf1rsowGAG6Xq0gAsKwON9V
co1M/Qgh9fbizx5PJx3uIWRM3SwD0QEojRkLSikmnwdrc08OA2Eo3zN9kNzfpDaS
CfbI81Qr
=u23E
-----END PGP SIGNATURE-----
```


</details>
